<launch>

    <arg name="device" default="0"/>
    <arg name="show" default="false"/>

    <node pkg="tuw_uvc" type="general_node" name="camera" output="screen">
        <param name="video_device" value="/dev/video$(arg device)"/>
        <param name="show_camera_image" value="$(arg show)"/>
        <param name="convert_image" value="1"/>
        <param name="camera_info_url" value="file://$(find tuw_uvc)/cfg/logitech.yml"/>
        <param name="frame_id" value="camera"/>
    </node>


    <node pkg="tuw_aruco" type="aruco_node" name="arMarkerAruco" output="screen">
        <param name="pose_estimation_enabled" value="false"/>
        <param name="publish_markers" value="false"/>
        <param name="publish_fiducials" value="true"/>

        <remap from="image" to="/camera/image_raw"/>
        <remap from="camera_info" to="/camera/camera_info"/>
        <remap from="markers" to="/markersAruco"/>
        <remap from="fiducials" to="/fiducialsAruco"/>
    </node>

    <node pkg="tuw_marker_pose_estimation" type="pose_estimation_node" name="arPoseEstimation" output="screen">
        <remap from="fiducials" to="/fiducialsAruco"/>
        <remap from="markers" to="/markersPoseEstimation"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find tuw_marker_pose_estimation)/rviz/demo_config.rviz" />

</launch>
